types/tx_batch.go

```diff
@@ -1,11 +1,21 @@
 package types
 
-import abci "github.com/tendermint/tendermint/abci/types"
+import (
+	"github.com/cosmos/cosmos-sdk/store/multiversion"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
 
 // DeliverTxEntry represents an individual transaction's request within a batch.
 // This can be extended to include tx-level tracing or metadata
 type DeliverTxEntry struct {
-	Request abci.RequestDeliverTx
+	Request  abci.RequestDeliverTx
+	Metadata DeliverTxMetadata
```

Comment:
2 cents: I'd probably would have just put the Writeset here, because Metadata normally means things like headers/trackers/tracers and not as much real-arguments to the function.